File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleFullAuditedEntityDto.cs
using System;
using Volo.Abp.Auditing;
using Volo.Abp.Data;

namespace Volo.Abp.Application.Dtos;

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
[Serializable]
public abstract class ExtensibleFullAuditedEntityDto<TPrimaryKey> : ExtensibleAuditedEntityDto<TPrimaryKey>, IFullAuditedObject
{
     * <inheritdoc />
    public boolean IsDeleted;// { get; set; }

     * <inheritdoc />
    public ID DeleterId;// { get; set; }

     * <inheritdoc />
    public DateTime? DeletionTime;// { get; set; }

    protected ExtensibleFullAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleFullAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
[Serializable]
public abstract class ExtensibleFullAuditedEntityDto : ExtensibleAuditedEntityDto, IFullAuditedObject
{
     * <inheritdoc />
    public boolean IsDeleted;// { get; set; }

     * <inheritdoc />
    public ID DeleterId;// { get; set; }

     * <inheritdoc />
    public DateTime? DeletionTime;// { get; set; }

    protected ExtensibleFullAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleFullAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

File: ueueo-frameworks/ueueo-ddd-application/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleFullAuditedEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.auditing.IFullAuditedObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * This class can be inherited by DTO classes to implement <see cref="IFullAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
 */
@EqualsAndHashCode(callSuper = true)
@Data
public abstract class ExtensibleFullAuditedEntityDto extends ExtensibleAuditedEntityDto implements IFullAuditedObject {

    private boolean isDeleted;

    private ID deleterId;

    private Date deletionTime;

    protected ExtensibleFullAuditedEntityDto() {
        this(true);
    }

    protected ExtensibleFullAuditedEntityDto(boolean setDefaultsForExtraProperties) {
        super(setDefaultsForExtraProperties);
    }

}
